tests/e2e_tests.py

```diff
@@ -0,0 +1,352 @@
+from models.output import Output
+from generators.model_generator import ModelGenerator
+from send_generated_input_script.rabbit_mq_publisher import RabbitMqPublisher
+from unittest import TestCase
+from tests.test_base import TestBase
+
+
+class E2ETest(TestCase, TestBase):
+
+    def setUp(self):
+        self.rabbitmq_publisher = RabbitMqPublisher()
+        os.chdir(TestBase.STUDENTS_DIR_PATH)
+        if os.path.exists(TestBase.STUDENTS_DIR_PATH):
+            for file in os.listdir(TestBase.STUDENTS_DIR_PATH):
+                os.remove(os.path.join(TestBase.STUDENTS_DIR_PATH, file))
+
+    def test_output_parse(self):
+        input_model = ModelGenerator.generate_input_model()
+        parsed_output = TestBase.parse_output(input_model)
+
+        TestBase.send_body(self.rabbitmq_publisher, input_model)
+
+        output_model = Output(**TestBase.read_from_file(f"{TestBase.STUDENTS_DIR_PATH}/{TestBase.get_docs(1)[0]}"))
+
+        self.assertEqual(output_model.studentDetails, parsed_output.studentDetails.__dict__)
+        for subject in range(len(parsed_output.subjectGrades)):
+            self.assertEqual(output_model.subjectGrades[subject], parsed_output.subjectGrades[subject].__dict__)
+        self.assertEqual(output_model.totalAvg, parsed_output.totalAvg)
+        self.assertEqual(output_model.birthDate, parsed_output.birthDate)
+        self.assertEqual(output_model.age, parsed_output.age)
+        self.assertEqual(output_model.gender, parsed_output.gender)
+        self.assertEqual(output_model.isGoodBehaviour, parsed_output.isGoodBehaviour)
+        self.assertEqual(output_model.age, input_model.age)
+        self.assertEqual(output_model.studentDetails.firstName, input_model.studentDetails['firstName'])
+        self.assertEqual(output_model.studentDetails.lastName, input_model.studentDetails['lastName'])
+        self.assertEqual(output_model.studentDetails.id, input_model.studentDetails['id'])
+
+    def test_text_file_saved_success(self):
+        input_model = ModelGenerator.generate_input_model()
+
+        TestBase.send_body(self.rabbitmq_publisher, input_model)
+
+        self.assertEqual(1, len(TestBase.get_docs(1)))
+        self.assertTrue(os.listdir(TestBase.STUDENTS_DIR_PATH) == [f"{input_model.studentDetails['firstName']}_{input_model.studentDetails['lastName']}.txt"])
+
+    def test_send_same_body_twice(self):
+        input_model = ModelGenerator.generate_input_model()
+
+        TestBase.send_body(self.rabbitmq_publisher, input_model)
+        TestBase.send_body(self.rabbitmq_publisher, input_model)
+
+        self.assertTrue(1, len(TestBase.get_docs(1)))
+
+    def test_two_text_files_saved_success(self):
+        input_model_1 = ModelGenerator.generate_input_model()
+        input_model_2 = ModelGenerator.generate_input_model()
+
+        TestBase.send_body(self.rabbitmq_publisher, input_model_1)
+        TestBase.send_body(self.rabbitmq_publisher, input_model_2)
+
+    def test_send_non_json_body(self):
```

Comment:
test_send_non_json_body_failure or test_sent_invalid_format_failure
you forgot the failure